extensions/amp-youtube/0.1/amp-youtube.js

```diff
@@ -186,15 +191,41 @@ class AmpYoutube extends AMP.BaseElement {
 
     this.iframe_ = iframe;
 
+    this.unlistenMessage_ = listen(
+      this.win,
+      'message',
+      this.handleYoutubeMessages_.bind(this)
+    );
     this.win.addEventListener(
-        'message', event => this.handleYoutubeMessages_(event));
-
+      'message', event => this.handleYoutubeMessages_(event)
+    );
     return this.loadPromise(iframe)
         .then(() => this.listenToFrame_())
         .then(() => this.playerReadyPromise_);
   }
 
   /** @override */
+  unlayoutCallback() {
+    if (this.iframe_) {
+      removeElement(this.iframe_);
+      this.iframe_ = null;
+    }
+    if (this.unlistenMessage_) {
+      this.unlistenMessage_();
+    }
+    if (this.playerState_ !== PlayerStates.PAUSED) {
+      this.playerState_ = PlayerStates.PAUSED;
+    }
+    if (this.playerReadyPromise_) {
+      this.playerReadyPromise_ = null;
```

Comment:
@jridgewell both with regard to the above comment and the next, are you thinking something along the lines of what's done on line ~122:
```js
this.playerReadyPromise_ = new Promise(resolve => {
  this.playerReadyResolver_ = resolve;
});
```